standalone-metastore/src/main/java/org/apache/hadoop/hive/metastore/txn/TxnHandler.java

```diff
@@ -577,6 +577,25 @@ public OpenTxnsResponse openTxns(OpenTxnRequest rqst) throws MetaException {
           LOG.debug("Going to execute update <" + q + ">");
           stmt.execute(q);
         }
+
+        if (rqst.isSetReplPolicy()) {
+          List<String> rowsRepl = new ArrayList<>();
+          String selectRepl = "select target_txn_id from TXN_MAP where repl_policy = " + quoteString(rqst.getReplPolicy()) + "and src_txn_id = " + rqst.getReplSrcTxnId();
+          for (long i = first; i < first + numTxns; i++) {
+            rowsRepl.add(quoteString(rqst.getReplPolicy()) + "," + rqst.getReplSrcTxnId() + "," + i);
+          }
+          List<String> queriesRepl = sqlGenerator.createInsertValuesStmt(
+                  "TXN_MAP (repl_policy, src_txn_id, target_txn_id)", rowsRepl);
+          for (String q : queriesRepl) {
+            rs = stmt.executeQuery(selectRepl);
```

Comment:
I think, this query is there to ensure open_txn event is idempotent. Need to see if this can be checked in server itself.

Reply:
if checked at server then we need to send extra message from hive server to meta store to check for txn validity